src/app/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts

```diff
@@ -1,35 +1,287 @@
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+        )),
+        map(([outcome, data]) =>
+          this.outcomeBehaviourProvider.isVisible(outcome,
+            'safetyAndBalance',
+            safetyAndBalanceQuestionsExist(data.safetyAndBalanceQuestions))),
+      ),
+      modeOfTransport$: currentTest$.pipe(
+        select(getTestSummary),
+        select(getModeOfTransport),
+      ),
+      displayDrivingFaultComments$: currentTest$.pipe(
+        select(getTestData),
+        withLatestFrom(
+          testCategory$,
+        ),
+        map(([data, category]) => this.shouldDisplayDrivingFaultComments(data, category)),
+      ),
+      safetyAndBalanceQuestions$: currentTest$.pipe(
+        select(getTestData),
+        select(getSafetyAndBalanceQuestions),
+        map((checks) => [...checks.safetyQuestions, ...checks.balanceQuestions]),
+      ),
+    };
+
+    this.setupSubscriptions();
+  }
+
+  setupSubscriptions() {
+    const {
+      startDateTime$,
+    } = this.pageState;
+    this.subscription = merge(
+      startDateTime$.pipe(map((value) => this.startDateTime = value)),
+    ).subscribe();
+  }
+
+  ionViewDidLeave(): void {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  setIsValidStartDateTime(isValid: boolean) {
+    this.isValidStartDateTime = isValid;
+  }
 
-  ngOnInit() {
+  dateOfTestChanged(inputDate: string) {
+    const customStartDate = getNewTestStartTime(inputDate, this.startDateTime);
+    this.store$.dispatch(TestStartDateChanged(this.startDateTime, customStartDate));
+    this.store$.dispatch(SetStartDate(customStartDate));
   }
 
-  navigateBack(): void {
-    this.navController.back();
+  identificationChanged(identification: Identification): void {
+    this.store$.dispatch(IdentificationUsedChanged(identification));
   }
 
-  async navigateForward(): Promise<void> {
-    await this.routeByCat.navigateToPage(TestFlowPageNames.REKEY_REASON_PAGE);
+  independentDrivingChanged(independentDriving: IndependentDriving): void {
+    this.store$.dispatch(IndependentDrivingTypeChanged(independentDriving));
   }
 
-  async navigateJournal(): Promise<void> {
-    await this.router.navigate([JOURNAL_PAGE]);
+  modeOfTransportChanged(modeOfTransport: ModeOfTransport): void {
+    this.store$.dispatch(ModeOfTransportChanged(modeOfTransport));
   }
 
+  weatherConditionsChanged(weatherConditions: WeatherConditions[]): void {
+    this.store$.dispatch(WeatherConditionsChanged(weatherConditions));
+  }
+
+  routeNumberChanged(routeNumber: number) {
+    this.store$.dispatch(RouteNumberChanged(routeNumber));
+  }
+
+  candidateDescriptionChanged(candidateDescription: string) {
+    this.store$.dispatch(CandidateDescriptionChanged(candidateDescription));
+  }
+
+  additionalInformationChanged(additionalInformation: string): void {
+    this.store$.dispatch(AdditionalInformationChanged(additionalInformation));
+  }
```

Comment:
I would imagine this could all be removed as they will exist in the office base page